src/plugins/intel_gpu/src/graph/impls/onednn/utils.cpp

```diff
@@ -374,17 +373,30 @@ cldnn::format find_data_format(dnnl::memory::desc desc) {
     if (onednn_desc != dnnl::memory::format_tag::undef) {
         return convert_data_format(onednn_desc);
     } else {
+        std::vector<format> search_fmt = {
```

Comment:
Please rework the logic for onednn memory::desc conversion to cldnn format tag to get rid of map and use matching by blocking desc and axes order.

Reply:
Applied.

Reply:
Implementation to get rid of convert_data_format and get_format_by_desc is reverted.
This will be handled by a separated PR.
There is an issue that information in mem::desc can not distinguish byfx and byxf if stride is [10, 1, 10, 1].

Reply:
In that case several tags have the same memory desc, so we can pick any tag (first match I suppose). Then we can just try to ensure that `layout::compatible` and `layout::identical` methods can detect that those layouts are actually same and we don't need to reorder the data. I.e. we may have an additional reorder byfx <-> byxf, but it'll be optimized out, since memory buffer doesn't require physical memory movements.

Reply:
Ok. I will apply a method to check layouts are actually same.

Reply:
Is it fine we apply this to another PR?
I will apply it to both find_data_format and find_format.